commands/utils/quick-pick-azure.ts

```diff
@@ -0,0 +1,168 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { Registry } from 'azure-arm-containerregistry/lib/models';
+import { ResourceGroup } from 'azure-arm-resource/lib/resource/models';
+import { Location, Subscription } from 'azure-arm-resource/lib/subscription/models';
+import * as opn from 'opn';
+import * as vscode from "vscode";
+import { skus } from '../../constants'
+import { ext } from '../../extensionVariables';
+import { IAzureQuickPickItem } from '../../node_modules/vscode-azureextensionui';
+import * as acrTools from '../../utils/Azure/acrTools';
+import { AzureImage } from "../../utils/Azure/models/image";
+import { Repository } from "../../utils/Azure/models/repository";
+import { AzureUtilityManager } from '../../utils/azureUtilityManager';
+
+export async function quickPickACRImage(repository: Repository, prompt?: string): Promise<AzureImage> {
+    const placeHolder = prompt ? prompt : 'Choose image to use';
+    const repoImages: AzureImage[] = await acrTools.getImagesByRepository(repository);
+    const imageListNames = repoImages.map(img => <IAzureQuickPickItem<AzureImage>>{ label: img.tag, data: img });
+    let desiredImage = await ext.ui.showQuickPick(imageListNames, { 'canPickMany': false, 'placeHolder': placeHolder });
+    return desiredImage.data;
+}
+
+export async function quickPickACRRepository(registry: Registry, prompt?: string): Promise<Repository> {
+    const placeHolder = prompt ? prompt : 'Choose registry to use';
+    const repositories: Repository[] = await acrTools.getRepositoriesByRegistry(registry);
+    const quickPickRepoList = repositories.map(repo => <IAzureQuickPickItem<Repository>>{ label: repo.name, data: repo });
+    let desiredRepo = await ext.ui.showQuickPick(quickPickRepoList, { 'canPickMany': false, 'placeHolder': placeHolder });
+    return desiredRepo.data;
+}
+
+export async function quickPickACRRegistry(canCreateNew: boolean = false, prompt?: string): Promise<Registry> {
+    const placeHolder = prompt ? prompt : 'Choose registry to use';
+    let registries = await AzureUtilityManager.getInstance().getRegistries();
```

Comment:
You're not passing in a sort function, there's the list is not sorted.  Really this is a problem with AzureUtilityManager.getRegistries - it should either:
1) require a sort function (not optional)
2) have a default sort if none is provided
3) or don't ask for a sort function, just always do a sort by name

The current implementation makes bugs like these likely.